sda-download/api/sda/sda.go

```diff
@@ -305,3 +347,32 @@ var sendStream = func(reader *streaming.Crypt4GHReader, writer http.ResponseWrit
 
 	return nil
 }
+
+// Calculates the start and end coordinats to use. If a range is set in HTTP headers,
+// it will be used as is. If not, the functions parameters will be used,
+// and adjusted to match the data block boundaries of the encrypted file.
+var calculateEncryptedCoords = func(start, end int64, htsget_range string, fileDetails *database.FileDownload) (int64, int64) {
+	if htsget_range != "" {
+		startEnd := strings.Split(strings.TrimPrefix(htsget_range, "bytes="), "-")
+		if len(startEnd) > 1 {
+			a, errA := strconv.ParseInt(startEnd[0], 10, 64)
+			b, errB := strconv.ParseInt(startEnd[1], 10, 64)
+			if errA == nil && errB == nil && a < b {
+
+				return a, b
+			}
+		}
+	}
+	// Adapt end coordinate to follow the crypt4gh block boundaries
+	headlength := bytes.NewReader(fileDetails.Header)
+	bodyEnd := int64(fileDetails.ArchiveSize)
+	if end > 0 {
+		var packageSize float64 = 65564 // 64KiB+28, 28 is for chacha20_ietf_poly1305
```

Comment:
@pontus @blankdots is the packageSize robust enough, or should the `+28` be more variable?

Reply:
Good enough for now. In the future, we should probably offer this as convenience functions in crypt4gh (where you'd call with a file header and get back e.g. this size).